src/main/java/org/folio/processing/mapping/MappingManager.java

```diff
@@ -110,13 +112,20 @@ private static void updateLocationsInMappingProfile(MappingProfile mappingProfil
    * @param mappingParameters - mapping parameters
    */
   private static void updateOrganizationsInMappingProfile(MappingProfile mappingProfile, MappingParameters mappingParameters) {
+    if (mappingParameters.getOrganizations() == null) {
+      return;
+    }
+
     if ((mappingProfile.getMappingDetails() != null) && (mappingProfile.getMappingDetails().getMappingFields() != null)) {
       HashMap<String, String> organizations = getOrganizationsFromMappingParameters(mappingParameters);
+      HashMap<String, String> donorOrganizations = getDonorOrganizationsFromMappingParameters(mappingParameters);
       if (!organizations.isEmpty()) {
         for (MappingRule mappingRule : mappingProfile.getMappingDetails().getMappingFields()) {
           if ((mappingRule.getName() != null) && (mappingRule.getName().equals(VENDOR_ID)
             || mappingRule.getName().equals(MATERIAL_SUPPLIER) || mappingRule.getName().equals(ACCESS_PROVIDER))) {
             mappingRule.setAcceptedValues(organizations);
+          } else if (mappingRule.getName().equals(DONOR_ORGANIZATION_IDS)) {
```

Comment:
@RuslanLavrov  Please add here `mappingRule.getName() != null`. Otherwise - there will be NPE in case of some invalid rules.

Reply:
updated thanks